docs/sections/user_guide/rocoto_yaml.rst

```diff
@@ -0,0 +1,606 @@
+.. _defining_a_workflow:
+
+Defining a Rocoto Workflow
+==========================
+
+:rocoto:`Rocoto<>` is a workflow manager widely used by :ufs:`UFS<>` users and developers. It uses a custom XML language to define a set of tasks, their computational resource needs on a batch system, and their dependencies. 
+
+To date, it has been challenging to manage XML files that must support a multitude of workflow options. The ``uw rocoto`` tool defines a YAML language that can be easily manipulated like any other key/value configuration file and translates it into the syntax required by Rocoto. This enables flexibility to arbitrarily make changes to any workflow while UFS Apps can track the highest priority configurations for Community and Operational use.
+
+
+Workflow Section
+~~~~~~~~~~~~~~~~
+Starting at the top level of the YAML config for Rocoto, there are several required sections. See the example and explanation below:
+
+.. code::
+
+  workflow:
+    attrs:
+      realtime: false
+      scheduler: slurm
+    cycledef:
+      - attrs:
+          activation_offset: -06:00
+          group: howdy
+        spec: 202209290000 202209300000 06:00:00
+    entities:
+      ACCOUNT: myaccount
+      FOO: test.log
+    log: /some/path/to/&FOO;
+    tasks:
+      ...
+
+YAML Entries
+............
+
+``attrs``: This section allows users to define any number of the available attributes available to the workflow tag in Rocoto's native XML language. Set them with a key/value that matches the exact syntax needed by Rocoto. These attributes work to fill in the ``<workflow>`` tag. For example,
+
+.. code:: XML
+
+  <workflow realtime="false" scheduler="slurm">
+  ...
+  </workflow>
+
+``cycledef``: This section is comprised of a list of grouped cycle definitions for a workflow. Any number of entries is supported. Similar to ``attrs`` for the ``workflow`` level, this entry, too has an ``attrs`` tag that follows the exact requirements of those in the Rocoto XML language. The ``spec`` entry is required and supports either the "start, stop, step" syntax, or the "crontab-like" method supported by Rocoto. The example above translates to a single ``<cycledef>`` tag:
+
+.. code:: XML
+
+  <cycledef group="howdy">202209290000 202209300000 06:00:00</cycledef>
+
+``entities``: This section is a set of key/value pairs where the keys are the named entities (or XML variables) and the values track to the values of those XML entities. Each entry shows up as an entry like ``<!ENTITY key "value">``. The example above would yield:
```

Comment:
- "track to" -> "are" -- "track to" sounds vague to me. I suspect that maybe you don't like the repetition of "values" so close together. What about something like
> This section maps named entities (i.e. XML variables), expressed as keys, to their values, with each pair rendered as `<!ENTITY key "value">` in the final XML.
?

Reply:
Ooh! I like that.